isaaclab_arena_environments/tabletop_gearmesh_environment.py

```diff
@@ -0,0 +1,143 @@
+                func=randomize_object_serials_pose,
+                mode="reset",
+                params={
+                    "pose_range": {"x": (0.25, 0.6), "y": (-0.20, 0.20), "z": (0.0, 0.0), "yaw": (-1.0, 1.0)},
+                    "min_separation": 0.18,
+                    "asset_cfgs": [SceneEntityCfg("gear_base"), SceneEntityCfg("medium_gear")],
+                    "relative_asset_cfgs": [SceneEntityCfg("small_gear"), SceneEntityCfg("large_gear")],
+                },
+            )
+
+        # Get assets from registry
+        gear_base = self.asset_registry.get_asset_by_name("gear_base")()
+        medium_gear = self.asset_registry.get_asset_by_name("medium_gear")()
+        small_gear = self.asset_registry.get_asset_by_name("small_gear")()
+        large_gear = self.asset_registry.get_asset_by_name("large_gear")()
+        background = self.asset_registry.get_asset_by_name(args_cli.background)()
+        light_spawner_cfg = sim_utils.DomeLightCfg(color=(0.75, 0.75, 0.75), intensity=500.0)
+        light = self.asset_registry.get_asset_by_name("light")(spawner_cfg=light_spawner_cfg)
+        embodiment = self.asset_registry.get_asset_by_name(args_cli.embodiment)(enable_cameras=args_cli.enable_cameras)
+        embodiment.scene_config.robot = FRANKA_PANDA_FACTORY_HIGH_PD_CFG.replace(prim_path="{ENV_REGEX_NS}/Robot")
+
+        if args_cli.teleop_device is not None:
+            teleop_device = self.device_registry.get_device_by_name(args_cli.teleop_device)()
+        else:
+            teleop_device = None
+
+        # Set initial poses for all 4 gears
+        # Based on reference FactoryGearMeshSceneCfg configuration
+        gear_base.set_initial_pose(
+            Pose(
+                position_xyz=(0.6, 0.0, 0.0),  # Gear base position
+                rotation_wxyz=(1.0, 0.0, 0.0, 0.0),
+            )
+        )
+
+        medium_gear.set_initial_pose(
+            Pose(
+                position_xyz=(0.5, 0.2, 0.0),  # Medium gear to be assembled
+                rotation_wxyz=(1.0, 0.0, 0.0, 0.0),
+            )
+        )
+
+        small_gear.set_initial_pose(
+            Pose(
+                position_xyz=(0.6, 0.0, 0.0),  # Small reference gear
+                rotation_wxyz=(1.0, 0.0, 0.0, 0.0),
+            )
+        )
+
+        large_gear.set_initial_pose(
+            Pose(
+                position_xyz=(0.6, 0.0, 0.0),  # Large reference gear
+                rotation_wxyz=(1.0, 0.0, 0.0, 0.0),
+            )
+        )
+
+        # Create scene with all 4 gears and background
+        scene = Scene(assets=[background, gear_base, medium_gear, small_gear, large_gear, light])
+
+        # Create gear mesh task
+        task = FactoryAssemblyTask(
+            fixed_asset=gear_base,
+            held_asset=medium_gear,
+            assist_asset_list=[small_gear, large_gear],
+            background_scene=background,
+        )
+        task.events_cfg = EventCfgGearMesh()
```

Comment:
We replace the event_cfg in the underlying task. Do we ever use the original event_cfg defined in `FactoryAssemblyTask`?